compiler/pyproject.toml

```diff
@@ -3,6 +3,10 @@ requires = [
     "setuptools>=42",
     "wheel",
     "cmake",
+    # Note that the compiler wheel does not presently need nanobind, but
+    # it's build is enabled by the same flag which enables the runtime
+    # configuration, which does.
+    "nanobind>=1.4.0",
```

Comment:
Maintaining two different dependencies / styles of defining bindings will take extra effort. Are you thinking about porting the compiler bindings to nanobind also? Maybe we'd need to keep using pybind for compiler bindings as long as upstream MLIR is using it too?

Are there any compatibility concerns about using one project for the runtime bindings and another for the compiler bindings?

Reply:
I think the MLIR side will be pinned to pybind11 for a very long time, so is what it is. We hardly have any pybind that we don't inherit from MLIR, and in a followup, I'll probably just remove our submodule in favor of the same style of pip-dependency for both.

There are no compatibility concerns or really any issue in mixing across such extensions. The MLIR python bindings were made to interop via C as a hermetic unit vs magical pybind specific stuff, so should just be an implementation detail.

The runtime bindings are performance sensitive and need to be optimized as such. The compiler bindings we inherit are not so much.